app/main.py

```diff
@@ -1,2 +1,57 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.threshold = int(self.capacity * 2 / 3)
+        self.hash_table = self.create_hash_table()
+
+    def create_hash_table(self) -> list:
+        return [[] for _ in range(self.capacity)]
+
+    def resize_capacity(self) -> None:
+        this_hash_table = self.hash_table
+        self.capacity *= 2
+        self.hash_table = self.create_hash_table()
+        self.length = 0
+        self.threshold = int(self.capacity * 2 / 3)
+
+        for item in this_hash_table:
+            if item:
+                self.__setitem__(item[0], item[2])
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.threshold == self.length:
+            self.resize_capacity()
+
+        hash_value = hash(key)
+        hash_index = hash_value % self.capacity
+
+        while True:
+            if not self.hash_table[hash_index]:
+                self.hash_table[hash_index] = [key, hash_value, value]
+                self.length += 1
+                break
+            if (self.hash_table[hash_index][0] == key
+                    and self.hash_table[hash_index][1] == hash_value):
```

Comment:
it's better to compare hashes first because comparing integers could be mush faster than comparing complex objects such as long tuples